app/build.gradle

```diff
@@ -59,6 +60,13 @@ dependencies {
     implementation 'com.intuit.ssp:ssp-android:1.1.0'
     implementation 'com.intuit.sdp:sdp-android:1.1.0'
 
+    // Retrofit Networking
+    implementation 'com.squareup.retrofit2:retrofit:2.9.0'
+    implementation 'com.squareup.retrofit2:converter-gson:2.9.0'
```

Comment:
I'd recommend to use anything but Gson for JSON parsing. 
Gson is a Java library and has its issues with things like nullability and default values in Kotlin (for example, it can assign null values to nullable fields). 

Checkout Moshi. Works really well with Retrofit ;)

Reply:
I implemented Moshi but it asks me to writ custom adapters for every Data class I have. It doesn't do it by default.

I know it has benefits however I am confident with Gson

Reply:
I'll target a new PR specially for this ::thumbs up::

Reply:
No need to make custom adapters. Instead of using @SerializedName , just use: @field:Json

Reply:
Tried this as well, still it gave adapter error everytime :(

Will try to do things again from scratch in the next PR